src/test/java/io/confluent/examples/streams/CustomJoinWithTableTriggeringStreamTableJoinIntegrationTest.java

```diff
@@ -0,0 +1,429 @@
+      //          -----+-----------------------------------+------------------------
+      //          10   | ("alice", 999.99)                 |
+      //          20   |                     ("alice", 1L) | ("alice", (999.99, 1L))
+      //          30   |                     ("alice", 2L) | ("alice", (999.99, 2L))
+      //
+      //       In the wrong example above, only one join output must be produced, not two.  It's up to you to decide
+      //       which one, however.
+      //
+      new KeyValue<>("alice", new Pair<>(999.99, 2L)),
+      new KeyValue<>("bob", new Pair<>(888.88, null))
+    );
+
+    //
+    // Step 1: Configure and start the processor topology.
+    //
+    final StreamsBuilder builder = new StreamsBuilder();
+
+    final Properties streamsConfiguration = new Properties();
+    streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, "table-trigger-join-integration-test");
+    streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+    streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.ByteArray().getClass().getName());
+    streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+    streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+    // Use a temporary directory for storing state, which will be automatically removed after the test.
+    streamsConfiguration.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getAbsolutePath());
+
+    // This state store is used to temporarily buffer any records arriving at the stream side of the join, so that
+    // we can wait (if needed) for matching data to arrive at the table side.
+    final StoreBuilder<KeyValueStore<String, Pair<Double, Instant>>> streamBufferStateStore =
+      Stores
+        .keyValueStoreBuilder(
+          Stores.persistentKeyValueStore("stream-buffer-state-store"),
+          Serdes.String(),
+          new PairSerde<>(Serdes.Double(), new InstantSerde())
+        )
+        .withCachingEnabled();
+    builder.addStateStore(streamBufferStateStore);
+
+    // Read the input data.
+    final KStream<String, Double> stream = builder.stream(inputTopicForStream, Consumed.with(Serdes.String(), Serdes.Double()));
+    final KTable<String, Long> table =
+      builder.table(inputTopicForTable, Consumed.with(Serdes.String(), Serdes.Long()), Materialized.as(tableStoreName));
+
+    // Perform the custom join operation.
+    final Duration maxWaitTimePerRecordForTableData = Duration.ofSeconds(5);
+    final Duration frequencyToCheckForExpiredWaitTimes = Duration.ofSeconds(2);
+    final KStream<String, Pair<Double, Long>> transformedStream =
+      stream.transform(
+        new StreamTableJoinStreamWaitsForTable(
+          maxWaitTimePerRecordForTableData,
+          frequencyToCheckForExpiredWaitTimes,
+          streamBufferStateStore.name(), tableStoreName),
```

Comment:
nit: `tableStoreName` should go to it's own line

Reply:
Ack